src/player.h

```diff
@@ -1370,6 +1371,7 @@ class player : public Character
         int radiation;
         unsigned long cash;
         int movecounter;
+        bool death_drops;// Turned to false for simulating NPCs on distant missions so they don't drop all their gear in sight
```

Comment:
If only NPCs use it and `player`s don't, it shouldn't be in `player`.
Instead, you should mark the `place_corpse` function as `virtual` (as in, `virtual bool place_corpse`, only in header), then create an override in `npc`, like this:
```
// npc.h
class npc {
    ...
    void place_corpse() override;
    ...
}
// npc.cpp
void npc::place_corpse() {
    if( death_drops ) {
        player::place_corpse();
    }
}
```

Also, you forgot to save and load it in savegames.

Reply:
Ah, ok.  Yup, don't need it to be saved since it is turned on and off in the method that utilizes it so I don't accidentally leave it on for NPCs or monsters.

Reply:
> don't need it to be saved since it is turned on and off in the method that utilizes it

Then you should have it in the function, not pass it around in the object itself.
Have a set or vector of NPCs that need death processing at specific spot. Or a map if the position changes.

Reply:
Hmmm... so it's an interrupter or whatever you call it, for combat deaths.  The only thing I call for actual combat is:
    const auto att = random_entry( attacker );
    monster *def = random_entry( group );
    att->melee_attack( *def, false);
Everything is handled by melee_attack() including dropping the corpse in the player om without the interrupter.  Would it be best to add another bool argument to melee_attack() that would get pushed to the monster getting hit which would go though whatever else till it arrived to its die() function?

Reply:
This won't work well: NPCs and monsters still have local coords and `melee_attack` will trigger local-coord-dependent behaviors.
Special casing every single event isn't feasible. You'll need to drop the actual combat mechanics and replace them with approximations.

Otherwise there will be a lot of strange bugs like
* Appearing items from destroyed NPC weapons and armor
* Acid bursts from acid defense when hitting some zombies
* Blood splatters from critters that NPCs are fighting

The actual proper solution would be to drop the `g` pointer and have NPCs and monsters carry their own `game` and `map` pointers and simulate their actions on those "virtual" maps. But this would be a lot of work.

There is no easy way of splitting off common sections of complex combat sections, so approximations are the only sensible way of doing this at the moment.